sqlglot/schema.py

```diff
@@ -206,84 +262,40 @@ def _normalize(self, schema: t.Dict) -> t.Dict:
 
         return normalized_mapping
 
-    def add_table(
-        self, table: exp.Table | str, column_mapping: t.Optional[ColumnMapping] = None
-    ) -> None:
-        """
-        Register or update a table. Updates are only performed if a new column mapping is provided.
+    def _normalize_table(self, table: exp.Table) -> exp.Table:
+        normalized_table = table.copy()
+        for arg in TABLE_ARGS:
+            value = normalized_table.args.get(arg)
+            if isinstance(value, (str, exp.Identifier)):
+                value = value.name if isinstance(value, exp.Identifier) else value
+                normalized_table.set(arg, self._normalize_name(value))
 
-        Args:
-            table: the `Table` expression instance or string representing the table.
-            column_mapping: a column mapping that describes the structure of the table.
-        """
-        table_ = self._ensure_table(table)
-        column_mapping = ensure_column_mapping(column_mapping)
-        schema = self.find(table_, raise_on_missing=False)
-
-        if schema and not column_mapping:
-            return
-
-        _nested_set(
-            self.mapping,
-            list(reversed(self.table_parts(table_))),
-            column_mapping,
-        )
-        self.mapping_trie = self._build_trie(self.mapping)
+        return normalized_table
 
     def _normalize_name(self, name: str) -> str:
         try:
-            identifier: t.Optional[exp.Expression] = sqlglot.parse_one(
-                name, read=self.dialect, into=exp.Identifier
+            identifier = t.cast(
+                exp.Identifier, sqlglot.parse_one(name, read=self.dialect, into=exp.Identifier)
             )
         except:
-            identifier = exp.to_identifier(name)
```

Comment:
why was this changed? what case triggered the need to change this

Reply:
There wasn't a case that triggered this change, but it felt reasonable to return `name` here if we are unable to parse it into an identifier. I guess there are two cases to consider here. If `name` can't be parsed and it is converted `to_identifier`, then the resulting `exp.Identifier` is:

- quoted, in which case we'd return `name` (this behaviour is preserved in this PR), or
- not quoted, so we'd return `name.lower()`.

How we deal with the latter case is the only difference here. The question is: why should we convert to an identifier and call `.lower` to begin with, if the name couldn't be parsed as one? Seemed more logical to just return `name`, but I might be wrong. Happy to revert this change if it looks risky.

FWIW, in my original PR Barak had suggested using `parse_one` here too (https://github.com/tobymao/sqlglot/pull/891#discussion_r1062872629), but I can see how `to_identifier` might make this more conservative.